connaisseur/alert.py

```diff
@@ -0,0 +1,125 @@
+import os
+import json
+import logging
+from datetime import datetime
+from string import Template
+from enum import Enum
+import requests
+from connaisseur.exceptions import AlertSendingError
+from connaisseur.mutate import get_container_specs
+from connaisseur.policy import ImagePolicy
+from connaisseur.image import Image
+
+
+class Alert:
+    """
+    Class to store image information about an alert as attributes and a sending functionality as method.
+    Alert Sending can, depending on the configuration, throw an AlertSendingError causing Connaisseur
+    responding with status code 500 to the request that was sent for admission control,
+    causing a Kubernetes Error event.
+    """
+
+    connaisseur_pod_id: str
+    cluster: str
+    timestamp: int
+
+    request_id: str
+    images: list
+
+    template: str
+    receiver_url: str
+    payload: dict
+    headers: dict
+
+    alert_message: str
+    priority: int
+
+    def __init__(self, alert_message, receiver_config, admission_request):
+        self.alert_message = alert_message
+        self.admission_request = admission_request
+        self.receiver_url = receiver_config["receiver_url"]
+        self.throw_if_alert_sending_fails = receiver_config[
+            "fail_if_alert_sending_fails"
+        ]
+        self.payload = self._construct_payload(receiver_config)
+        self.headers = get_headers(receiver_config)
+        self.template = receiver_config["template"]
+
+    def _construct_payload(self, receiver_config):
+        with open(
+            "{}/{}".format(os.getenv("ALERT_CONFIG_DIR"), receiver_config["template"]),
+            "r",
+        ) as templatefile:
+            template = str(templatefile.read())
+            payload = json.loads(template)
```

Comment:
may raise some JSON exception. What then? - It's fine to have it bubble up, just wondered if it was intentional.

Reply:
As it's validated against a `jsonschema` in the first place, we prevent this kind of error here